node/src/ledger/ledger_manager.rs

```diff
@@ -0,0 +1,515 @@
+use mina_p2p_messages::v2::{
+    LedgerHash, MinaBaseAccountBinableArgStableV2, MinaBaseSparseLedgerBaseStableV2,
+    MinaLedgerSyncLedgerAnswerStableV2, MinaLedgerSyncLedgerQueryStableV1,
+    MinaStateProtocolStateValueStableV2, NonZeroCurvePoint, StateHash,
+};
+use openmina_core::block::ArcBlockWithHash;
+use openmina_core::channels::{mpsc, oneshot};
+use openmina_core::snark::{Snark, SnarkJobId};
+use std::collections::{BTreeMap, BTreeSet};
+use std::sync::{Arc, Mutex};
+
+use super::ledger_service::LedgerCtx;
+use crate::account::AccountPublicKey;
+use crate::block_producer::vrf_evaluator::DelegatorTable;
+use crate::block_producer::{BlockProducerWonSlot, StagedLedgerDiffCreateOutput};
+use crate::ledger::LedgerAddress;
+use crate::p2p::channels::rpc::StagedLedgerAuxAndPendingCoinbases;
+use crate::rpc::RpcScanStateSummaryScanStateJob;
+use crate::transition_frontier::sync::{
+    ledger::staged::StagedLedgerAuxAndPendingCoinbasesValid,
+    TransitionFrontierRootSnarkedLedgerUpdates,
+};
+use crate::transition_frontier::CommitResult;
+
+/// The type enumerating different requests that can be made to the
+/// service. Each specific constructor has a specific response
+/// constructor associated with it. Unfortunately, this relationship
+/// can't be expressed in the Rust type system at the moment. For this
+/// reason this type is private while functions wrapping the whole call
+/// to the service are exposed as the service's methods.
+enum LedgerRequest {
+    AccountsSet {
+        snarked_ledger_hash: LedgerHash,
+        parent: Arc<Mutex<LedgerAddress>>,
```

Comment:
Lets just use a LedgerAddress here, there is no advantage in wrapping it in `Arc` and `Mutex` right? it is a cheap value to copy, no need to share it.